workspaces/orchestrator/plugins/orchestrator/src/api/OrchestratorClient.ts

```
@@ -104,10 +111,38 @@ export class OrchestratorClient implements OrchestratorApi {
     const defaultApi = await this.getDefaultAPI();
     const reqConfigOption: AxiosRequestConfig =
       await this.getDefaultReqConfig();
+    // We know the injected ScmIntegrationsApi instance is actually a ScmIntegrationRegistry with .list()
+    const integrations = (this.scmIntegrationsApi as any).list?.();
+    const authTokens: { provider: string; token: string }[] = [];
+    for (const integration of integrations) {
+      const provider = integration.type;
+      const url = `https://${integration.config.host}`;
+      if (!url) continue;
+      try {
+        const credentials = await this.scmAuthApi.getCredentials({
+          url,
+          optional: false,
+        });
+
+        if (credentials?.token) {
+          authTokens.push({
+            provider,
+            token: credentials.token,
+          });
+        }
+      } catch (e) {
+        console.warn(`No token available for ${provider}`, e);
```

no need to re-throw the error?

---

@gciavarrini no need to re-throw the error here as the backend-plugin will itself throw 400 bad error if no auth tokens are present. We just want to silently log in the error.